libs/sql-parser/src/main/antlr/SqlBase.g4

```diff
@@ -537,11 +537,12 @@ rerouteOption
     ;
 
 dataType
-    : definedDataType           #definedDataTypeDefault
-    | ident                     #dataTypeIdent
-    | objectTypeDefinition      #objectDataType
-    | ARRAY '(' dataType ')'    #arrayDataType
-    | dataType '[]'             #arrayDataType
+    : definedDataType                                          #definedDataTypeDefault
+    | ident                                                    #dataTypeIdent
+    | objectTypeDefinition                                     #objectDataType
+    | ARRAY '(' dataType ')'                                   #arrayDataType
+    | dataType '[]'                                            #arrayDataType
+    | dataType '(' integerLiteral (',' integerLiteral )* ')'   #parametrizedDataType
```

Comment:
It is a bit looking into the future, but we would need more than one parameter for some data types.

Reply:
> It is a bit looking into the future, but we would need more than one parameter for some data types, so I decided to it now.

Sounds good.

---

Is this valid for objects and arrays? Otherwise we could restrict it to the `definedDataType` and `ident` cases.